napalm_eos/eos.py

```diff
@@ -1201,3 +1201,143 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
+        if neighbor_address:
+            peer_info = self._parse_per_peer_bgp_detail(
+                raw_output[0]['output'])
+
+            # Appending peer_info to final data structure
+            if int(peer_info[0]['remote_as']) not in bgp_detail_info:
+                bgp_detail_info[int(peer_info[0]['remote_as'])] = []
+
+            (bgp_detail_info[int(peer_info[0]['remote_as'])].
+                append(peer_info[0]))
+
+        else:
+            # Using preset template to extract peer info
+            v4_peer_info = self._parse_per_peer_bgp_detail(
+                raw_output[0]['output'])
+
+            v6_peer_info = self._parse_per_peer_bgp_detail(
+                raw_output[1]['output'])
+
+            for peer_info in v4_peer_info:
+                # Appending v4 peer_info to final data structure
+                if int(peer_info['remote_as']) not in bgp_detail_info:
+                    bgp_detail_info[int(peer_info['remote_as'])] = []
+
+                bgp_detail_info[int(peer_info['remote_as'])].append(peer_info)
+
+            for peer_info in v6_peer_info:
+                # Appending v6 peer_info to final data structure
+                if int(peer_info['remote_as']) not in bgp_detail_info:
+                    bgp_detail_info[int(peer_info['remote_as'])] = []
+
+                bgp_detail_info[int(peer_info['remote_as'])].append(peer_info)
+
+        return bgp_detail_info
+
+    def _parse_per_peer_bgp_detail(self, peer_output):
+        """This function parses the raw data per peer and returns a
+        json structure per peer.
+        """
+
+        int_fields = ['accepted_prefix_count', 'local_as', 'remote_as',
+                      'local_port', 'remote_port', 'local_port',
+                      'input_messages', 'output_messages', 'input_updates',
+                      'output_updates', 'messages_queued_out', 'holdtime',
+                      'configured_holdtime', 'keepalive',
+                      'configured_keepalive', 'advertise_prefix_count',
+                      'received_prefix_count']
+
+        peer_details = []
+
+        # Using preset template to extract peer info
+        peer_info = (
+            napalm_base.helpers.textfsm_extractor(
+                self, 'bgp_detail', peer_output))
+
+        for item in peer_info:
+            # Issuing command to get number of accepted prefixes for peer
+            try:
+                peer_ver = IPAddress(item['remote_address']).version
+            except Exception:
+                continue
+
+            if peer_ver == 4:
+                acc_prefix_command = (
+                    'show ip bgp summary | include %s' %
+                    item['remote_address'])
+            elif peer_ver == 6:
+                acc_prefix_command = (
+                    'show ipv6 bgp summary | include %s' %
+                    item['remote_address'])
+
+            accepted_prefix_output = (
+                self.device.run_commands([acc_prefix_command],
```

Comment:
You are right, this information is not available in the main view, but only in ```sh ip bgp sum```. Bad news.
For only one single field (pretty important but not terrible at the end of the day), my feeling is that it does not worth issuing so many requests.
It is faster, but still very slow:

```python
>>> start = time.time(); d = e.get_bgp_neighbors_detail(); time.time() - start
80.33029413223267
```

I would propose to remove this section and set the value to -1.

But let's wait for @dbarrosop's opinion first.